### ISSUE TITLE
Better documentation snippets

### ISSUE THREAD

**Issue by keathley** [elixir-lang/language-elixir#54]
Right now the documentation snippets aren't very complete and aren't very usable. Ideally we could create a snippet like:

```coffeescript
  'moduledoc':
    'prefix': 'moduledoc'
    'body': """
      @moduledoc \"\"\"
      ${0:documentation}
      \"\"\"
    """
```

But unfortunately this is currently not supported in atom. There's an open issue on this here: atom/snippets#140. Based on the resolution of that issue I'd like to rework the documentation snippets to be more consistent.

**Comment by gusaiani** [comment]
Just an observation, `\"""` seems to work just as well as `\"\"\"`.

**Comment by keathley** [comment]
Good catch. Unfortunately we're still stuck until that issue is resolved. I've thought about changing the snippet to something like:

```elixir
@doc""" # <- the cursor would go here
"""
```

That way the user can hit enter to expand the doc. Its not an ideal solution but it would provide better support until atom works out a fix upstream.

**Comment by gusaiani** [comment]
Seems just fine for the time being.

**Comment by gusaiani** [comment]
This seems to work, apparently even for nested modules:
```
  'moduledoc':
    'prefix': 'moduledoc'
    'body': """
      @moduledoc \"""\n\b\b${0:documentation}\n\"""
    """
```

**Comment by keathley** [comment]
Thats awesome!!! Would you want to submit a PR with with these fixes? If not thats fine. I can try to get to it a bit later.

**Comment by gusaiani** [comment]
Glad you liked it :)
If you can point me to which snippets you're planning to add or fix, I'll be happy to give them a shot.
It's getting late here, I'll be back tomorrow.
If you'd rather go ahead and tackle them, all the merrier.

Thanks always for this package.

**Comment by keathley** [comment]
Lets just start with the `doc` and `moduledoc` snippets. I think that we're missing snippets for almost all of the attributes (`@callback`, `@type`, `@spec`, etc.) but we can handle that in another PR.

**Comment by keathley** [comment]
Closed with #62 and #63
